sklearn/decomposition/_kernel_pca.py

```diff
@@ -44,19 +45,20 @@ class KernelPCA(TransformerMixin, BaseEstimator):
         Independent term in poly and sigmoid kernels.
         Ignored by other kernels.
 
-    kernel_params : mapping of string to any, default=None
-        Parameters (keyword arguments) and values for kernel passed as
-        callable object. Ignored by other kernels.
+    kernel_params : dict, default=None
+        Mapping of string to any. Parameters (keyword arguments) and
+        values for kernel passed as callable object.
+        Ignored by other kernels.
 
-    alpha : int, default=1.0
+    alpha : float, default=1.0
         Hyperparameter of the ridge regression that learns the
         inverse transform (when fit_inverse_transform=True).
 
     fit_inverse_transform : bool, default=False
         Learn the inverse transform for non-precomputed kernels.
         (i.e. learn to find the pre-image of a point)
 
-    eigen_solver : string ['auto'|'dense'|'arpack'], default='auto'
+    eigen_solver : {'auto', 'dense', 'arpack'}, str, default='auto'
```

Comment:
we don't need to specify the type if we already give the complete set of allowed values

```suggestion
    eigen_solver : {'auto', 'dense', 'arpack'}, default='auto'
```